pagination/paginator.go

```diff
@@ -0,0 +1,118 @@
+package pagination
+
+import (
+	"fmt"
+	"math"
+)
+
+const defaultPageNumber = 1
+const defaultItemsPerPage = 10
+
+// PaginatorInterface defines what makes a paginator
+type PaginatorInterface interface {
```

Comment:
Interfaces should not have the word `Interface` in golang. Generally it should be a description of what it is doing (which would imply `Paginator`). I guess that means that the struct would become `Pagination`